src/Data/Text/Utf8.hs

```diff
@@ -262,7 +262,8 @@ skipCodePointsBackwards (Text !u8data !off !len) !index0 !n0
     loop index n | atTrailingByte index =
       loop (index-1) n  -- Don't exit before we're at a leading byte
     loop index 0 | index < 0 =
-      -- We don't prevent reads outside the array, but you do get an error afterwards
+      -- Throw an error if we've read before the array, this one-time check
+      -- doesn't prevent segfaults but may help you locate bugs.
```

Comment:
`s/segfaults/undefined behavior` - If you get a segfault, you won't see this check firing.

Also, I'd add an "(e.g. when the data was not valid UTF-8)" to make clear that this is not expected.